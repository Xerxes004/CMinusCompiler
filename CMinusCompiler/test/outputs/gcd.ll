(FUNCTION  gcd  [(int u) (int v)]
  (BB 2
    (OPER 3 Func_Entry []  [])
  )
  (BB 3
    (OPER 5 Mov [(r 4)]  [(i 0)])
    (OPER 4 NEQ [(r 3)]  [(r 2)(r 4)])
    (OPER 6 BEQ []  [(r 3)(i 0)(bb 5)])
  )
  (BB 4
    (OPER 7 Mov [(m RetReg)]  [(r 1)])
  )
  (BB 6
  )
  (BB 1
    (OPER 1 Func_Exit []  [])
    (OPER 2 Return []  [(m RetReg)])
  )
  (BB 5
    (OPER 9 Pass []  [(r 2)])
    (OPER 12 Div_I [(r 7)]  [(r 1)(r 2)])
    (OPER 11 Mul_I [(r 6)]  [(r 7)(r 2)])
    (OPER 10 Sub_I [(r 5)]  [(r 1)(r 6)])
    (OPER 13 Pass []  [(r 5)])
    (OPER 14 JSR []  [(s gcd)])
    (OPER 15 Mov [(m RetReg)]  [(r 8)])
    (OPER 16 Mov [(m RetReg)]  [(r 8)])
    (OPER 18 Jmp []  [(bb 6)])
  )
)
(FUNCTION  main  []
  (BB 2
    (OPER 3 Func_Entry []  [])
  )
  (BB 3
    (OPER 4 JSR []  [(s input)])
    (OPER 5 Mov [(m RetReg)]  [(r 3)])
    (OPER 6 Mov [(r 1)]  [(r 3)])
    (OPER 7 JSR []  [(s input)])
    (OPER 8 Mov [(m RetReg)]  [(r 4)])
    (OPER 9 Mov [(r 2)]  [(r 4)])
    (OPER 10 Pass []  [(r 1)])
    (OPER 11 Pass []  [(r 2)])
    (OPER 12 JSR []  [(s gcd)])
    (OPER 13 Mov [(m RetReg)]  [(r 5)])
    (OPER 14 Pass []  [(r 5)])
    (OPER 15 JSR []  [(s output)])
    (OPER 16 Mov [(m RetReg)]  [(r 6)])
  )
  (BB 1
    (OPER 1 Func_Exit []  [])
    (OPER 2 Return []  [(m RetReg)])
  )
)
