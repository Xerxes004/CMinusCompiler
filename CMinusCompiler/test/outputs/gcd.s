.text
	.align 4
.globl  gcd
gcd:
gcd_bb2:
	pushq	%R15
	movl	%EDI, %ECX
	movl	%ESI, %EDI
gcd_bb3:
	movl	$0, %EAX
	cmpl	%EAX, %EDI
	je	gcd_bb5
gcd_bb4:
	movl	%ECX, %EAX
gcd_bb1:
	popq	%R15
	ret
gcd_bb5:
	movl	$0, %EDX
	movl	%ECX, %EAX
	idivl	%EDI, %EAX
	imull	%EDI, %EAX
	movl	%ECX, %EDI
	subl	%EAX, %EDI
	movl	%EDI, %ESI
	call	gcd
	movl	%R15D, %EAX
	movl	%R15D, %EAX
	jmp	gcd_bb1
.globl  main
main:
main_bb2:
	pushq	%R12
	pushq	%R13
	pushq	%R14
	pushq	%R15
main_bb3:
	call	input
	movl	%R12D, %EAX
	call	input
	movl	%R13D, %EAX
	movl	%R13D, %EAX
	movl	%R12D, %EDI
	movl	%EAX, %ESI
	call	gcd
	movl	%R14D, %EAX
	movl	%R14D, %EDI
	call	output
	movl	%R15D, %EAX
main_bb1:
	popq	%R15
	popq	%R14
	popq	%R13
	popq	%R12
	ret
