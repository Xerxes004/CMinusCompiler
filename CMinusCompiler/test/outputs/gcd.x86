(FUNCTION  gcd  [(int u) (int v)]
  (BB 2
    (OPER 19 Mov [(r 1)]  [(m EDI)])
    (OPER 20 Mov [(r 2)]  [(m ESI)])
  )
  (BB 3
    (OPER 5 Mov [(r 4)]  [(i 0)])
    (OPER 21 Cmp [(m Flags)]  [(r 2)(r 4)])
    (OPER 22 BEQ []  [(bb 5)(m Flags)])
  )
  (BB 4
    (OPER 7 Mov [(m EAX)]  [(r 1)])
  )
  (BB 1
    (OPER 2 Return []  [(m EAX)])
  )
  (BB 5
    (OPER 23 Mov [(m EDI)]  [(r 2)])
    (OPER 24 Mov [(m EDX)]  [(i 0)])
    (OPER 25 Mov [(m EAX)]  [(r 1)])
    (OPER 12 Div [(m EAX)(m EDX)]  [(m EAX)(r 2)(m EDX)])
    (OPER 26 Mov [(r 7)]  [(m EAX)])
    (OPER 27 Mov [(m EAX)]  [(r 7)])
    (OPER 11 Mul [(m EAX)(m EDX)]  [(m EAX)(r 2)])
    (OPER 28 Mov [(r 6)]  [(m EAX)])
    (OPER 29 Mov [(r 5)]  [(r 1)])
    (OPER 10 Sub_I [(r 5)]  [(r 5)(r 6)])
    (OPER 30 Mov [(m ESI)]  [(r 5)])
    (OPER 14 JSR []  [(s gcd)])
    (OPER 15 Mov [(m EAX)]  [(r 8)])
    (OPER 16 Mov [(m EAX)]  [(r 8)])
    (OPER 18 Jmp []  [(bb 1)])
  )
)
(FUNCTION  main  []
  (BB 2
  )
  (BB 3
    (OPER 4 JSR []  [(s input)])
    (OPER 5 Mov [(m EAX)]  [(r 3)])
    (OPER 6 Mov [(r 1)]  [(r 3)])
    (OPER 7 JSR []  [(s input)])
    (OPER 8 Mov [(m EAX)]  [(r 4)])
    (OPER 9 Mov [(r 2)]  [(r 4)])
    (OPER 17 Mov [(m EDI)]  [(r 1)])
    (OPER 18 Mov [(m ESI)]  [(r 2)])
    (OPER 12 JSR []  [(s gcd)])
    (OPER 13 Mov [(m EAX)]  [(r 5)])
    (OPER 19 Mov [(m EDI)]  [(r 5)])
    (OPER 15 JSR []  [(s output)])
    (OPER 16 Mov [(m EAX)]  [(r 6)])
  )
  (BB 1
    (OPER 2 Return []  [(m EAX)])
  )
)
