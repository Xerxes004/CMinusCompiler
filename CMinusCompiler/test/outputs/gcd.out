                 INT	int
                  ID	gcd
              LPAREN	(
                 INT	int
                  ID	u
               COMMA	,
                 INT	int
                  ID	v
              RPAREN	)
         LCURLYBRACE	{
                  IF	if
              LPAREN	(
                  ID	v
           NOT_EQUAL	!=
                 NUM	0
              RPAREN	)
         LCURLYBRACE	{
              RETURN	return
                  ID	u
           SEMICOLON	;
         RCURLYBRACE	}
                ELSE	else
         LCURLYBRACE	{
              RETURN	return
                  ID	gcd
              LPAREN	(
                  ID	v
               COMMA	,
                  ID	u
               MINUS	-
                  ID	u
              DIVIDE	/
                  ID	v
            MULTIPLY	*
                  ID	v
              RPAREN	)
           SEMICOLON	;
         RCURLYBRACE	}
         RCURLYBRACE	}
                VOID	void
                  ID	main
              LPAREN	(
                VOID	void
              RPAREN	)
         LCURLYBRACE	{
                 INT	int
                  ID	x
           SEMICOLON	;
                 INT	int
                  ID	y
           SEMICOLON	;
                  ID	x
              ASSIGN	=
                  ID	input
              LPAREN	(
              RPAREN	)
           SEMICOLON	;
                  ID	y
              ASSIGN	=
                  ID	input
              LPAREN	(
              RPAREN	)
           SEMICOLON	;
                  ID	output
              LPAREN	(
                  ID	gcd
              LPAREN	(
                  ID	x
               COMMA	,
                  ID	y
              RPAREN	)
              RPAREN	)
           SEMICOLON	;
         RCURLYBRACE	}
